src/partials/messenger.ts

```diff
@@ -844,6 +854,26 @@ class ConversationController {
         const chat = this.domChatElement;
         this.showScrollJump = chat.scrollHeight - (chat.scrollTop + chat.offsetHeight) > 10;
     }
+
+    /**
+     * Mark the current conversation as pinned.
+     */
+    public pinConversation(): void {
+        this.webClientService
+            .modifyConversation(this.conversation, true)
+            .then(() => this.showMessage('messenger.PINNED_CONVERSATION_OK'))
+            .catch((msg) => this.$log.error(this.logTag, 'Pinning conversation failed: ' + msg));
```

Comment:
I'm wondering if we should also show toast messages for these kinds of responses.

Pro: It clearly is a response to the user's action and let's it know that something it requested didn't work.
Con: It is a bug if it doesn't work and showing the toast as well as adding translations for it is a bit of overhead.

Reply:
Yep, but at least there is some feedback, versus none at all.

Reply:
My opinion is that we should either provide UI feedback along the line of *this particular thing didn't work* or we don't provide any UI feedback. It's helpful for the user to know that i.e. pinning a conversation didn't work but *oops, something didn't work but I can't tell you what* is not in any way helpful and would just confuse the user. Note that I would not consider showing a generic error an issue if there wasn't a noticeable delay between user action. But since we send messages back and forth, there is.

Reply:
If the user clicks the "pin" button, it's better if nothing happens and there's an "an error occurred" message 2 seconds later, than if simply nothing happens. The user will click again and again.

By showing a generic error message, this allows the user to actually describe something to support, which could then ask to take a look at the debug log. If we provide no feedback at all, the user will say "pinning does not work", which is very unclear.

But of course, if you think it's important we could create a specific error message for this condition and localize it, although it's unlikely that this will occur. There's not much that can go wrong when pinning (except bugs of course).

Reply:
Mhh, that's a good point. The user will eventually figure out that this toast will have something to do with it furiously clicking the button. :grin: 

I wasn't referring to this particular rejection though but rather a question on whether we should do this for rejections in general. If we add toast messages for rejections, I think they should preferrably be specific and localised. That was my point.

Reply:
FIne, I localized the error messages in the latest fixup commit.